contents/docs/plugins/build.md

```diff
@@ -73,14 +72,81 @@ Most options in this file are self-explanatory, but there are a few worth explor
 
 The type of a parameter in the config can be either `string` or `attachment`. If the type is set to `attachment`, PostHog will prompt the user for a file upload during the configuration step.
 
+### PluginMeta
+
+> Check out [Plugin Types](/docs/plugins/types) for a full spec of types for plugin authors.
+
+**Every plugin server function** is called by the plugin server with an object of type `PluginMeta` that will always contain the object `cache`, and can also include `global`, `attachments`, and `config`, which you can use in your logic. 
+
+Here's what they do:
+
+#### config
+
+Gives you access to the plugin config values as described in `plugin.json` and configured via the PostHog interface.
+
+#### cache
+
+A way to store values that persist across `processEvent` calls. The values are stored in [Redis](https://redis.io/), an in-memory store.
+
+The `cache` type is defined as follows:
+
+```js
+interface CacheExtension {
+    set: (key: string, value: unknown, ttlSeconds?: number) => Promise<void>
+    get: (key: string, defaultValue: unknown) => Promise<unknown>
+    incr: (key: string) => Promise<number>
+    expire: (key: string, ttlSeconds: number) => Promise<boolean>
+}
+```
+
+Storing values is done via `cache.set`, which takes a key and a value, as well as an optional value in seconds after which the key will expire.
+
+Retrieving values uses `cache.get`, which takes the key of the value to be retrieved, as well as a default value in case the key does not exist.
+
+You can also use `cache.incr` to increment numerical values by 1, and `cache.expire` to make [keys volatile](https://redis.io/commands/expire), meaning they will expire after the specified number of seconds.
+
+Example usage:
+
+```js
+async function processEvent(event, {cache }) {
+  await cache.set('counter', 1) 
+  await cache.incr('counter') 
+  let counter = await cache.get('counter') // counter = 2
+  await cache.expire('counter', 60) // Expire counter after 1min
+
+  event['counter'] = counter
+  return event
+}
+```
+
+> **Note:** `processEvent` is addressed later in this doc.
+
+#### global
+
+Global is used for sharing functionality between `setupPlugin` and `processEvent` or `processEventBatch`, since global scope does not work in the context of PostHog plugins. 
+
+#### attachments
+
+Attachments gives access to files uploaded by the user for config parameters of type `attachment`. An `attachment` has the following type definition:
+
+```js
+interface PluginAttachment {
+    content_type: string
+    file_name: string
+    contents: any
```

Comment:
This is actually a `Buffer` in nodejs, even though written as `any` in the type.

I think I had to put it as `any`, as otherwise including the types would throw errors if used outside of the plugins, e.g. when importing types from `posthog-plugins` inside the posthog app itself.

Thus it would be good to mention below that this `attachments.attachmentName.contents` can be placed inside any function that accepts a NodeJS `Buffer`.